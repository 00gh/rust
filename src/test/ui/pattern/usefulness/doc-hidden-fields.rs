// aux-build:hidden.rs

extern crate hidden;

use hidden::HiddenStruct;

fn main() {
    let HiddenStruct { one, two, } = HiddenStruct::default();
    //~^ pattern requires `..` due to inaccessible fields

    let HiddenStruct { one, } = HiddenStruct::default();
    //~^ pattern does not mention field `two` and inaccessible fields

    let HiddenStruct { one, hide } = HiddenStruct::default();
    //~^ pattern does not mention field `two`
}
