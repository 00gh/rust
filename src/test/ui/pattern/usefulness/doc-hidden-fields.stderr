error: pattern requires `..` due to inaccessible fields
  --> $DIR/doc-hidden-fields.rs:8:9
   |
LL |     let HiddenStruct { one, two, } = HiddenStruct::default();
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: ignore the inaccessible and unused fields
   |
LL |     let HiddenStruct { one, two, .., } = HiddenStruct::default();
   |                                ++++

error[E0027]: pattern does not mention field `two` and inaccessible fields
  --> $DIR/doc-hidden-fields.rs:11:9
   |
LL |     let HiddenStruct { one, } = HiddenStruct::default();
   |         ^^^^^^^^^^^^^^^^^^^^^ missing field `two` and inaccessible fields
   |
help: include the missing field in the pattern and ignore the inaccessible fields
   |
LL |     let HiddenStruct { one, two, .. } = HiddenStruct::default();
   |                           ~~~~~~~~~~~
help: if you don't care about this missing field, you can explicitly ignore it
   |
LL |     let HiddenStruct { one, .. } = HiddenStruct::default();
   |                           ~~~~~~

error[E0027]: pattern does not mention field `two`
  --> $DIR/doc-hidden-fields.rs:14:9
   |
LL |     let HiddenStruct { one, hide } = HiddenStruct::default();
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^ missing field `two`
   |
help: include the missing field in the pattern
   |
LL |     let HiddenStruct { one, hide, two } = HiddenStruct::default();
   |                                 ~~~~~~~
help: if you don't care about this missing field, you can explicitly ignore it
   |
LL |     let HiddenStruct { one, hide, .. } = HiddenStruct::default();
   |                                 ~~~~~~

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0027`.
